nitransforms/base.py

```diff
@@ -251,14 +252,57 @@ class TransformBase:
     __slots__ = (
         "_reference",
         "_ndim",
+        "_affine",
+        "_shape",
+        "_header",
+        "_grid",
+        "_mapping",
```

Comment:
This is looking like TransformBase is becoming an affine transform. I will try to get it back to a generic, abstract base class.